tests/e2e/utils/CsrfExtractor.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+import { JSDOM } from 'jsdom';
+
+export class CsrfExtractor
+{
+    public extractToken(body: string): string
+    {
+        const dom = new JSDOM(body);
+        const input = dom.window.document.querySelector(
+            'input[name="_glpi_csrf_token"]'
+        ) as HTMLInputElement;
+
+        if (input === null || input.value === null) {
+            throw new Error(`Unexpected body content: ${body}`);
+        }
+
+        return input.value;
```

Comment:
All pages should contain a `<meta>` tag with the CSRF token.

```suggestion
        const meta = dom.window.document.querySelector(
            'meta[property="glpi:csrf_token"]'
        ) as HTMLMetaElement;

        if (meta === null || meta.content === null) {
            throw new Error(`Unexpected body content: ${body}`);
        }

        return meta.content;
```